htlcswitch/switch.go

```diff
@@ -676,7 +679,7 @@ func (s *Switch) IsForwardedHTLC(chanID lnwire.ShortChannelID,
 // given to forward them through the router. The sending link's quit channel is
 // used to prevent deadlocks when the switch stops a link in the midst of
 // forwarding.
-func (s *Switch) ForwardPackets(linkQuit <-chan struct{},
+func (s *Switch) ForwardPackets(ctx context.Context, linkQuit <-chan struct{},
```

Comment:
also need to check for ctx done when checking `<-linkQuit`

Reply:
good catch, thanks 🙏  (in future, we should be able to actually remove the `linkQuit` and just rely on the ctx i think)